src/PrestaShopBundle/Twig/Layout/TemplateVariablesBuilder.php

```diff
@@ -0,0 +1,145 @@
+
+namespace PrestaShopBundle\Twig\Layout;
+
+use PrestaShop\PrestaShop\Adapter\Feature\MultistoreFeature;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Context\EmployeeContext;
+use PrestaShop\PrestaShop\Core\Context\LegacyControllerContext;
+use PrestaShop\PrestaShop\Core\Context\ShopContext;
+use PrestaShop\PrestaShop\Core\FeatureFlag\FeatureFlagStateCheckerInterface;
+use PrestaShopBundle\Entity\Repository\TabRepository;
+use PrestaShopBundle\Entity\Tab;
+use PrestaShopBundle\Service\DataProvider\UserProvider;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+
+/**
+ * Allows you to construct variables used in rendering
+ */
+class TemplateVariablesBuilder
+{
+    public function __construct(
+        private readonly LegacyContext $context,
+        private readonly bool $debugMode,
+        private readonly RequestStack $requestStack,
+        private readonly CsrfTokenManagerInterface $tokenManager,
+        private readonly UserProvider $userProvider,
+        private readonly string $psVersion,
+        private readonly ConfigurationInterface $configuration,
+        private readonly MenuBuilder $menuBuilder,
+        private readonly TabRepository $tabRepository,
+        private readonly FeatureFlagStateCheckerInterface $featureFlagStateChecker,
+        private readonly EmployeeContext $employeeContext,
+        private readonly ShopContext $shopContext,
+        private readonly LegacyControllerContext $legacyControllerContext,
+        private readonly MultistoreFeature $multistoreFeature,
+    ) {
+    }
+
+    public function build(): TemplateVariables
+    {
+        return new TemplateVariables(
+            $this->context->getLanguage()->getIsoCode(),
+            $this->featureFlagStateChecker->isEnabled('symfony_layout'),
+            (bool) $this->context->getLanguage()->isRTL(),
+            $this->legacyControllerContext->controller_name,
+            $this->multistoreFeature->isActive(),
+            $this->isMenuCollapsed(),
+            $this->getJsRouterMetadata(),
+            $this->debugMode,
+            $this->installDirExists(),
+            $this->psVersion,
+            $this->getDefaultTabLink(),
+            $this->isMaintenanceEnabled(),
+            $this->isFrontOfficeAccessibleForAdmins(),
+            $this->isDisplayedWithTabs(),
+            $this->getBaseUrl(),
+        );
+    }
+
+    private function isMenuCollapsed(): bool
+    {
+        $cookie = $this->context->getContext()->cookie;
+
+        if (isset($cookie->collapse_menu)) {
+            return boolval($cookie->collapse_menu);
+        }
+
+        return false;
+    }
+
+    private function getJsRouterMetadata(): array
+    {
+        return [
+            // base url for javascript router
+            'base_url' => $this->requestStack->getCurrentRequest()->getBaseUrl(),
```

Comment:
Could this be replaced with?
```suggestion
            'base_url' => $this->shopContext->getBaseURL(),
```

Or are the two values different?

Reply:
I went to see, `$this->requestStack->getCurrentRequest()->getBaseUrl()` does not return at all what we actually want. It returns '/admin-dev/index.php', for what we want we should rather use `getSchemeAndHttpHost()`. Before we used the shop contex  and the `baseUrl` method, so let's stick with it and use the new context